Description: Set default device timeout to 8.0, otherwise graphics fails.
 The patch is needed because Ubuntu doesn't install the upstream configuration
 file defaults.plymouth (which contains DeviceTimeout=8), but uses
 alternatives instead.
Original-Author: Dimitri John Ledkov <xnox@ubuntu.com>
Author: Daniel van Vugt <daniel.van.vugt@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1838725
Forwarded: not-needed
Last-Update: 2020-04-01
---
 src/main.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: plymouth_0.9.4git20200323-0ubuntu6/src/main.c
===================================================================
--- plymouth_0.9.4git20200323-0ubuntu6.orig/src/main.c
+++ plymouth_0.9.4git20200323-0ubuntu6/src/main.c
@@ -2265,6 +2265,10 @@ main (int    argc,
         find_system_default_splash (&state);
         find_distribution_default_splash (&state);
 
+        /* Device timeout may not be NAN or zero */
+        if (isnan (state.device_timeout) || state.device_timeout <= 0.0)
+                state.device_timeout = 8.0;
+
         if (ply_kernel_command_line_has_argument ("plymouth.ignore-serial-consoles"))
                 device_manager_flags |= PLY_DEVICE_MANAGER_FLAGS_IGNORE_SERIAL_CONSOLES;
 
